### PR TITLE
Prune unverified users backend

### LINKED ISSUE #195: Prune unverified users (backend)
**Branch:** `prune-unverified-users-backend`

### Task at Hand

1. Make your changes in `helpers.ts` in the backend and also `server.ts`
2. Write a helper function that looks through Firebase and deletes every user that has an unverified email (and also our local database).
3. Update the cron job in `server.ts` to call your function every 24 hours.
4. Confirm that your code and pull request follow the [Pull Request Checklist](https://www.notion.so/h4i/Pull-Request-Checklist-910d5e6925df49569067f00f6aa0c0f3).### Acceptance Criteria

### Acceptance Criteria

- [ ] Unverified users are pruned every day in Firebase and our local database.

### Notes

- Potentially relevant stack overflow: https://stackoverflow.com/questions/67148672/how-to-delete-unverified-e-mail-addresses-in-firebase-authentication-flutter
- Use the Firebase Admin SDK documentation!

### PR DESCRIPTION
**Pull request by owenchen04**
## Summary

Added helper function `deleteUnverifiedUsers` in `helpers.ts` to delete users with an unverified email from firebase and the local database. Schedules the function to run in `server.ts` every 24 hours as a cron job. Only deletes unverfied users if they were created over 24 hours ago (to prevent users from being deleted immediately after they create their account but haven't yet verified it).

## Testing

We used manual testing on individual dummy users since we couldn't directly try deleting every unverified user from firebase (as most users are currently unverified). We started by console.log-ing every user and whether or not they're verified. We then picked a specific user and added an additional if-branch to just delete that user, checking to make sure they were deleted in both firebase and the local database.

## Notes

- **WARNING**: be careful when un-commenting code, as you might unintentionally delete most of the users in firebase if you just let it run. To test, switch the cron job in `server.ts` to run every minute, then pick a user to test on and add an if-branch inside `deleteUnverifiedUsers` to just delete that user.
- Closes #195

### CODE REVIEW

File: backend/src/utils/helpers.ts [hunk 1]
```diff
@@ -17,3 +21,50 @@ export const attempt = async (
...
+      if (!userData.emailVerified && timeDifference > twentyFourHours) {
+        
+        // await admin.auth().deleteUser(userData.uid);
+        
+        // await prisma.user.delete({
+        //   where: {
+        //     email: userData.email,
+        //   },
+        // });
+
```
Comment (akinfelami) [inline backend/src/utils/helpers.ts:59]: What happens if any of these throw an error?

**Review by akinfelami — CHANGES_REQUESTED** [review]
Hey guys!

Great work on this! Just one comment that we can chat about. Let me know!

**Review by akinfelami — APPROVED** [review]
Last commit to fix merge conflicts!

**Comment by akinfelami** [comment]
Okay, last comment here: 
Last introduction of websockets, introduced issues to the test suites. In additional to some other fixes, Got around to address them in this PR, and as you can see, checks pass now. 

Now, some of the other PRs are good to go, but they still fail the CI tests. I can cherry pick the commits from here into those PRs, to make sure checks pass and no merge conflicts, but I imagine they will get bloated. Lmk what you think!

**Review by jasozh — APPROVED** [review]

**Pull request merged**